src/app/modules/time-entries/pages/time-entries.component.ts

```diff
@@ -62,17 +62,40 @@ export class TimeEntriesComponent implements OnInit {
   }
 
   doSave(event: SaveEntryEvent) {
-    event.entry.start_date = new Date(event.entry.start_date).toISOString();
-    if (event.entry.end_date !== null && event.entry.end_date !== undefined) {
-      event.entry.end_date = new Date(event.entry.end_date).toISOString();
-    }
     if (this.entryId) {
+      const startDateChanged = this.entry.start_date !== event.entry.start_date;
+      const endDateChanged = this.entry.end_date !== event.entry.end_date;
+
+      if (startDateChanged) {
+        const startDate = new Date(event.entry.start_date);
```

Comment:
Lines 71, 72, 73 are the same than 89,90,91. Do not you think it would be great if we [extract them to a method](https://refactoring.guru/extract-method).

Reply:
Your're right. I thought the same. I just didn't know the code style to include functions. I will ask for help to Rene for this and also to fix the tests. Thanks for pointing it out.